webapp/channels/src/components/announcement_bar/notification_permission_bar/notification_permission_never_granted_bar.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+
+import {AnnouncementBarTypes} from 'utils/constants';
+import {requestNotificationPermission} from 'utils/notifications';
+
+export default function NotificationPermissionNeverGrantedBar() {
+    const [show, setShow] = useState(true);
+
+    const handleClick = useCallback(async () => {
+        await requestNotificationPermission();
+
+        // Dismiss the bar after user makes a choice
+        setShow(false);
+    }, []);
+
+    const handleClose = useCallback(() => {
+        // If the user closes the bar, don't show the notification bar any more for the rest of the session, but
+        // show it again on app refresh.
+        setShow(false);
+    }, []);
+
+    if (!show) {
+        return null;
+    }
+
+    return (
+        <AnnouncementBar
+            showCloseButton={true}
+            handleClose={handleClose}
+            type={AnnouncementBarTypes.ANNOUNCEMENT}
+            message={
+                <FormattedMessage
+                    id='announcementBar.notification.permissionNeverGrantedBar.message'
+                    defaultMessage='We need your permission to show web browser notifications.'
+                />
+            }
+            ctaText={
+                <FormattedMessage
+                    id='announcementBar.notification.permissionNeverGrantedBar.cta'
```

Comment:
I think this is fine because the context here has been changed

Reply:
I think the change of context only matters if the text actually changes, but I'm fine leaving it since I've definitely blocked this PR for long enough with my other text changes 😅